lib/rules/selector-not-notation/index.mjs

```diff
@@ -113,6 +114,7 @@ const rule = (primary) => {
 					node: ruleNode,
 					result,
 					ruleName,
+					line: range.start.line,
```

Comment:
Blocked until we resolve: https://github.com/stylelint/stylelint/pull/8086

-----

You shouldn't need to pass `line` to get the expected result.

Can you remove this after we resolve #8086 ?

-----

I also noticed that this rule uses a bit of custom code to calculate a range.

I don't think this is actually needed.
Bringing this rule in line with others by calculating an `index` and `endIndex` might be better.

This would also allow us to remove this custom code in this rule.

Reply:
> I also noticed that this rule uses a bit of custom code to calculate a range.
> I don't think this is actually needed.

It is. Ill explain when we will get there.
In short it's related to cloning and the loss of information.

Reply:
I tried with `index` and `endIndex` while workin on `8086` and that just worked 🤔
If there is complexity maybe it is better as a separate change. I don't want to derail @ryo-manba's work here :)

Reply:
We still can't remove the `line` argument because of : https://github.com/stylelint/stylelint/issues/8126

I tested `index` and `endIndex` again and this just works fine. (ignoring upstream bugs around serialization of `:not(a, )`)
Using `index` and `endIndex` allows for the removal of the `line` argument.

I'm focussing on this extra `line` argument because it reveals obvious bugs in our code for `report()`. The code to calculate the ranges is already unique to this rule and adding even more code to cover other bugs seems like an incorrect path to me.

For now however I think we can keep the `line` argument while we resolve surrounding issues?